CodeEdit/Breadcrumbs/BreadcrumbsView.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  BreadcrumbsView.swift
+//  CodeEdit
+//
+//  Created by Lukas Pistrol on 17.03.22.
+//
+
+import SwiftUI
+import WorkspaceClient
+
+struct BreadcrumbsView: View {
+
+	@ObservedObject var workspace: WorkspaceDocument
+	let file: WorkspaceClient.FileItem
+
+	@State private var projectName: String = ""
+	@State private var folders: [String] = []
+	@State private var fileName: String = ""
+	@State private var fileImage: String = "doc"
+
+	init(_ file: WorkspaceClient.FileItem, workspace: WorkspaceDocument) {
+		self.file = file
+		self.workspace = workspace
+	}
+
+	var body: some View {
+		ZStack(alignment: .leading) {
+			Rectangle()
+				.foregroundStyle(Color(nsColor: .controlBackgroundColor))
+			HStack {
+				BreadcrumbsComponent(projectName, systemImage: "square.dashed.inset.filled", color: .accentColor)
+				chevron
+				ForEach(folders, id:\.self) { folder in
+					BreadcrumbsComponent(folder, systemImage: "folder.fill")
+					chevron
+				}
+				BreadcrumbsComponent(fileName, systemImage: fileImage, color: .accentColor)
+			}
+			.padding(.leading, 12)
+		}
+		.frame(height: 29)
+		.overlay(alignment: .bottom) {
+			Divider()
+		}
+		.onAppear {
+			fileInfo()
```

Comment:
Why you get fileInfo in `onAppear` instead of `init`?

Reply:
Because the `@State` seems to get initialized after `.init()` and so the names are blank

Reply:
All right. Just feel weird do that in `onAppear`, it means every time the window is active it'll be called.